diff -Naur libexpat-R_2_2_5/expat/CMakeLists.txt expat_patch/expat/CMakeLists.txt
--- libexpat-R_2_2_5/expat/CMakeLists.txt	2017-11-01 10:47:55.000000000 -0400
+++ expat_patch/expat/CMakeLists.txt	2018-06-26 16:40:13.185314600 -0400
@@ -13,7 +13,7 @@
 if(WINCE)
     set(BUILD_tools_default OFF)
 else()
-    set(BUILD_tools_default ON)
+    set(BUILD_tools_default OFF)
 endif()
 if(MSVC OR NOT BUILD_tools_default)
     set(BUILD_doc_default OFF)
@@ -27,9 +27,9 @@
 endif()
 
 option(BUILD_tools "build the xmlwf tool for expat library" ${BUILD_tools_default})
-option(BUILD_examples "build the examples for expat library" ON)
-option(BUILD_tests "build the tests for expat library" ON)
-option(BUILD_shared "build a shared expat library" ON)
+option(BUILD_examples "build the examples for expat library" OFF)
+option(BUILD_tests "build the tests for expat library" OFF)
+option(BUILD_shared "build a shared expat library" OFF)
 option(BUILD_doc "build man page for xmlwf" ${BUILD_doc_default})
 option(USE_libbsd "utilize libbsd (for arc4random_buf)" OFF)
 option(INSTALL "install expat files in cmake install target" ON)
@@ -75,7 +75,7 @@
 if(FLAG_NO_STRICT_ALIASING)
     set(EXTRA_COMPILE_FLAGS "${EXTRA_COMPILE_FLAGS} -fno-strict-aliasing")
 endif()
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_COMPILE_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_COMPILE_FLAGS} -fPIC")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${EXTRA_COMPILE_FLAGS}")
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/lib)
