diff -Naur expressions.orig/CMakeLists.txt expressions/CMakeLists.txt
--- expressions.orig/CMakeLists.txt	2013-09-17 15:37:51.000000000 +0100
+++ expressions/CMakeLists.txt	2013-09-17 15:37:33.000000000 +0100
@@ -0,0 +1,5 @@
+project(expressions)
+
+cmake_minimum_required(VERSION 2.6)
+
+install(FILES AST.h Evaluator.h Exception.h expressions.h Generator.h Memory.h Parser.h Tokenizer.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/expressions)

