diff -Naur sphinx-orig/sphinx/domains/cpp.py sphinx/sphinx/domains/cpp.py
--- sphinx-orig/sphinx/domains/cpp.py	2010-12-31 11:26:48.000000000 +1100
+++ sphinx/sphinx/domains/cpp.py	2010-12-31 11:27:56.000000000 +1100
@@ -390,11 +390,13 @@
 
 class FuncDefExpr(NamedDefExpr):
 
-    def __init__(self, name, visibility, static, explicit, rv,
+    def __init__(self, name, visibility, static, extern, virtual, explicit, rv,
                  signature, const, pure_virtual):
         NamedDefExpr.__init__(self, name, visibility, static)
         self.rv = rv
         self.signature = signature
+        self.extern = extern
+        self.virtual = virtual
         self.explicit = explicit
         self.const = const
         self.pure_virtual = pure_virtual
@@ -409,6 +411,10 @@
 
     def __unicode__(self):
         buf = self.get_modifiers()
+        if self.extern:
+            buf.append(u'extern')
+        if self.virtual:
+            buf.append(u'virtual')
         if self.explicit:
             buf.append(u'explicit')
         if self.rv is not None:
@@ -761,6 +767,16 @@
 
     def parse_function(self):
         visibility, static = self._parse_visibility_static()
+        if self.skip_word('extern'):
+            extern = True
+            self.skip_ws()
+        else:
+            extern = False
+        if self.skip_word('virtual'):
+            virtual = True
+            self.skip_ws()
+        else:
+            virtual = False
         if self.skip_word('explicit'):
             explicit = True
             self.skip_ws()
@@ -774,8 +790,8 @@
             rv = None
         else:
             name = self._parse_type()
-        return FuncDefExpr(name, visibility, static, explicit, rv,
-                           *self._parse_signature())
+        return FuncDefExpr(name, visibility, static, extern, virtual, explicit,
+                           rv, *self._parse_signature())
 
     def parse_class(self):
         visibility, static = self._parse_visibility_static()
diff -Naur sphinx-orig/tests/test_cpp_domain.py sphinx/tests/test_cpp_domain.py
--- sphinx-orig/tests/test_cpp_domain.py	2010-12-31 11:26:49.000000000 +1100
+++ sphinx/tests/test_cpp_domain.py	2010-12-31 11:32:11.000000000 +1100
@@ -41,7 +41,12 @@
     assert unicode(parse('type_object', x)) == x
 
     assert unicode(parse('type_object', 'long long int foo')) == 'long long foo'
-
+    
+    x = 'virtual void myclass::foo(Bar& wee) const = 0'
+    assert unicode(parse('function', x)) == x
+    
+    x = 'extern void somefunc(const aPointer& thing)'
+    assert unicode(parse('function', x)) == x
 
 def test_operators():
     x = parse('function', 'void operator new [  ] ()')
