diff -Naur yaml-cpp-0.5.1.orig/CMakeLists.txt yaml-cpp-0.5.1/CMakeLists.txt
--- yaml-cpp-0.5.1.orig/CMakeLists.txt	2014-01-30 09:45:54.000000000 +0000
+++ yaml-cpp-0.5.1/CMakeLists.txt	2014-01-30 09:51:04.000000000 +0000
@@ -140,7 +140,7 @@
 		set(GCC_EXTRA_OPTIONS "${GCC_EXTRA_OPTIONS} ${FLAG_TESTED}")
 	endif()
 	#
-	set(CMAKE_CXX_FLAGS "-Wall ${GCC_EXTRA_OPTIONS} -pedantic -Wno-long-long ${CMAKE_CXX_FLAGS}")
+	set(CMAKE_CXX_FLAGS "-Wall ${GCC_EXTRA_OPTIONS} -pedantic -Wno-long-long -fPIC -fvisibility-inlines-hidden -fvisibility=hidden ${CMAKE_CXX_FLAGS}")
 	#
 	add_custom_target(debuggable $(MAKE) clean
 		COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Debug ${CMAKE_SOURCE_DIR}

--- src/emitterstate.cpp-orig	2015-01-30 13:53:26.468688052 +0000
+++ src/emitterstate.cpp	2015-01-30 13:53:33.507624388 +0000
@@ -267,7 +267,7 @@
 
     bool EmitterState::SetFloatPrecision(int value, FMT_SCOPE scope)
     {
-        if(value < 0 || value > std::numeric_limits<float>::digits10)
+        if(value < 0 || value > (2 + std::numeric_limits<float>::digits * 3010/10000))
             return false;
         _Set(m_floatPrecision, value, scope);
         return true;
@@ -275,7 +275,7 @@
     
     bool EmitterState::SetDoublePrecision(int value, FMT_SCOPE scope)
     {
-        if(value < 0 || value > std::numeric_limits<double>::digits10)
+        if(value < 0 || value > (2 + std::numeric_limits<double>::digits * 3010/10000))
             return false;
         _Set(m_doublePrecision, value, scope);
         return true;
