diff -Naur yaml-cpp-0.2.5/CMakeLists.txt YAML_CPP_LOCAL/CMakeLists.txt
--- yaml-cpp-0.2.5/CMakeLists.txt	2010-03-16 06:25:02.000000000 +1100
+++ YAML_CPP_LOCAL/CMakeLists.txt	2010-10-19 17:00:32.488681000 +1100
@@ -2,7 +2,7 @@
 
 project (YAML_CPP)
 
-set(LIB_TYPE SHARED)
+set(LIB_TYPE STATIC)
 
 if(IPHONE)
   set(CMAKE_OSX_SYSROOT iphoneos2.2.1)
@@ -57,6 +57,7 @@
 set_target_properties(yaml-cpp PROPERTIES
 	VERSION "${YAML_CPP_VERSION}"
 	SOVERSION "${YAML_CPP_VERSION_MAJOR}.${YAML_CPP_VERSION_MINOR}"
+    COMPILE_FLAGS "-fPIC"
 )
 
 install(TARGETS yaml-cpp ${_INSTALL_DESTINATIONS})
