diff -Naur yaml-cpp.orig/CMakeLists.txt yaml-cpp/CMakeLists.txt
--- yaml-cpp.orig/CMakeLists.txt	2012-01-21 03:52:48.000000000 -0500
+++ yaml-cpp/CMakeLists.txt	2017-10-11 14:13:47.762198400 -0400
@@ -235,10 +235,18 @@
 	${contrib_private_headers}
 )
 
+if(UNIX)
+    set(YAML_COMPILE_FLAGS "${YAML_COMPILE_FLAGS} -fPIC -fvisibility=hidden")
+endif()
+if(OCIO_INLINES_HIDDEN AND UNIX)
+    set(YAML_COMPILE_FLAGS "${YAML_COMPILE_FLAGS} -fvisibility-inlines-hidden")
+endif()
+
 set_target_properties(yaml-cpp PROPERTIES
 	VERSION "${YAML_CPP_VERSION}"
 	SOVERSION "${YAML_CPP_VERSION_MAJOR}.${YAML_CPP_VERSION_MINOR}"
 	PROJECT_LABEL "yaml-cpp ${LABEL_SUFFIX}"
+    COMPILE_FLAGS "${YAML_COMPILE_FLAGS}"
 )
 
 if(IPHONE)
diff -Naur yaml-cpp.orig/src/scanner.h yaml-cpp/src/scanner.h
--- yaml-cpp.orig/src/scanner.h	2012-01-21 03:52:48.000000000 -0500
+++ yaml-cpp/src/scanner.h	2017-10-11 14:14:10.420198400 -0400
@@ -57,7 +57,7 @@
 		
 		bool InFlowContext() const { return !m_flows.empty(); }
 		bool InBlockContext() const { return m_flows.empty(); }
-		int GetFlowLevel() const { return m_flows.size(); }
+		int GetFlowLevel() const { return int(m_flows.size()); }
 		
 		Token::TYPE GetStartTokenFor(IndentMarker::INDENT_TYPE type) const;
 		IndentMarker *PushIndentTo(int column, IndentMarker::INDENT_TYPE type);
