diff -Naur yaml-cpp-r423/CMakeLists.txt YAML_CPP_LOCAL/CMakeLists.txt
--- yaml-cpp-r423/CMakeLists.txt	2010-10-19 08:35:34.000000000 +1100
+++ YAML_CPP_LOCAL/CMakeLists.txt	2010-10-25 21:42:26.000000000 +1100
@@ -2,7 +2,7 @@
 
 project (YAML_CPP)
 
-set(LIB_TYPE SHARED)
+set(LIB_TYPE STATIC)
 
 if(IPHONE)
   set(CMAKE_OSX_SYSROOT iphoneos4.0)
@@ -58,6 +58,7 @@
 set_target_properties(yaml-cpp PROPERTIES
 	VERSION "${YAML_CPP_VERSION}"
 	SOVERSION "${YAML_CPP_VERSION_MAJOR}.${YAML_CPP_VERSION_MINOR}"
+    COMPILE_FLAGS "-fPIC -fvisibility-inlines-hidden -fvisibility=hidden"
 )
 
 if(IPHONE)
