diff -Naur yaml-cpp.orig/CMakeLists.txt yaml-cpp/CMakeLists.txt > yaml-cpp-0.3.0.patch
--- yaml-cpp.orig/CMakeLists.txt	2012-01-25 14:33:58.708744000 -0800
+++ yaml-cpp/CMakeLists.txt	2012-01-25 14:34:10.989085000 -0800
@@ -238,6 +238,7 @@
 set_target_properties(yaml-cpp PROPERTIES
 	VERSION "${YAML_CPP_VERSION}"
 	SOVERSION "${YAML_CPP_VERSION_MAJOR}.${YAML_CPP_VERSION_MINOR}"
+	COMPILE_FLAGS "-fPIC -fvisibility-inlines-hidden -fvisibility=hidden"
 	PROJECT_LABEL "yaml-cpp ${LABEL_SUFFIX}"
 )
 
